core/unit_test/CMakeLists.txt

```diff
@@ -1116,7 +1215,14 @@ if (NOT KOKKOS_HAS_TRILINOS)
       INPUT TestDeviceAndThreads.py
       ${USE_SOURCE_PERMISSIONS_WHEN_SUPPORTED}
     )
-    if(NOT Kokkos_ENABLE_OPENMPTARGET)  # FIXME_OPENMPTARGET does not select the right device
+    if(NOT Kokkos_ENABLE_OPENMPTARGET # FIXME_OPENMPTARGET does not select the right device
+       AND
+       # FIXME_OPENACC - does not select specified device beyond 0 until OpenACC
+       # backend is initialized.  For example, adding a Kokkos::parallel_for to
+       # the start of main in core/unit_test/UnitTest_DeviceAndThreads.cpp makes
+       # the test pass.
```

Comment:
I am not sure I understand your point.  Are you saying that `acc_set_device_num` does not have effect until a parallel region actually gets executed?
```C++
int dev_num = <some-non-zero-device-num>;
acc_set_device_num(dev_num);
assert(acc_get_device_num() == dev_num);  // fails

Reply:
The comment and condition-checking code are outdated; deleted.
The original issue was about the interaction between Kokkos initialization and the OpenACC backend implementation, which is fixed now.